### PR TITLE
Add initial Roadmap draft

### PR DESCRIPTION
**Pull request by jamesmunns**
As requested by ISRG, I've prepared a first draft of a feature-focused roadmap towards a 1.0 release.

### CODE REVIEW

**Comment by jamesmunns** [comment]
Note:

* Add a sentence wrt: prioritization for each marquee feature
* "Reason for importance for each major feature".

**Comment by jamesmunns** [comment]
Note:

* Pingora is interested in caching in the near future (next 6mo), but likely only the interface
* Pingora is interested in H3, based on prioritization

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+1. The application MUST support the use of the Automatic Certificate Management Environment (ACME)
+   protocol to obtain new TLS certificates.
+2. The application MUST support the use of ACME protocol to renew TLS certificates.
+3. The application MUST support the configuration of domain names to be managed (including obtaining
+   and renewal steps) automatically
+4. The application MUST support both fully qualified and wildcard domains.
+5. The application MUST support configuration of certificate renewal interval, from either:
+    1. The number of days since the certificate was acquired
+    2. The number of days until the certificate will expired
+6. The application MUST support API Version 2 of the ACME protocol
```
Comment (mcpherrinm) [inline docs/roadmap.md:77]: I think this should say RFC 8555 ACME.  "version 1" / "version 2" are let's encrypt specific names.  "version 1" doesn't exist anymore, effectively

**Comment by jamesmunns** [comment]
This was reviewed at the biweekly status meeting, and I'll merge now. Open to feedback in the future!

**Pull request merged**